### PR TITLE
Update the scheduling name for the created fusion and root instructions

### PR DESCRIPTION
**Pull request by zhenying-liu**
When working on maxtext, there is an error of scheduling name mismatching the instruction name for the remat-policy of minimal_offloaded. This CL fixes the scheduling names for the instructions created by the StreamAttributeAnnotator pass.

### CODE REVIEW

File: xla/service/gpu/stream_attribute_annotator.cc [hunk 1]
```diff
@@ -120,6 +120,13 @@ absl::StatusOr<bool> WrapIntoFusionAndAnnotateStreamAttributes(
       fusion_instruction->fused_instructions_computation(),
       absl::StrCat("wrapped_", wrapped_opcode, "_computation"));
   if (module->has_schedule()) {
+    // Update the scheduling names of the fusion and its root instruction
+    // to match their newly assigned instruction names during creation.
+    fusion_instruction->set_metadata_scheduling_name(
+        std::string(fusion_instruction->name()));
+    HloInstruction* root =
+        fusion_instruction->called_computations()[0]->root_instruction();
+    root->set_metadata_scheduling_name(std::string(root->name()));
```
Comment (akuegel) [inline xla/service/gpu/stream_attribute_annotator.cc:129]: maybe set_metadata_scheduling_name method can be changed to take a absl::string_view as argument?

File: xla/service/gpu/stream_attribute_annotator.cc [hunk 2]
```diff
@@ -120,6 +120,13 @@ absl::StatusOr<bool> WrapIntoFusionAndAnnotateStreamAttributes(
       fusion_instruction->fused_instructions_computation(),
       absl::StrCat("wrapped_", wrapped_opcode, "_computation"));
   if (module->has_schedule()) {
+    // Update the scheduling names of the fusion and its root instruction
+    // to match their newly assigned instruction names during creation.
+    fusion_instruction->set_metadata_scheduling_name(
+        std::string(fusion_instruction->name()));
+    HloInstruction* root =
+        fusion_instruction->called_computations()[0]->root_instruction();
```
[the remark this answers is not in the document]
Comment (zhenying-liu) [reply xla/service/gpu/stream_attribute_annotator.cc]: Done.

File: xla/service/gpu/stream_attribute_annotator.cc (hunk 1 above)
Comment (zhenying-liu) [reply xla/service/gpu/stream_attribute_annotator.cc:128]: set_metadata_scheduling_name() is declared and defined in hlo_instruction.h. If I change it, I'm afraid the change will be too big as at least another pass SchedulingInstructionAnnotator also uses it.

Comment (akuegel) [reply xla/service/gpu/stream_attribute_annotator.cc]: I checked and it is only used by that other pass. Anyway, this is a pre-existing issue, so I will take care of it separately.

Comment (akuegel) [reply xla/service/gpu/stream_attribute_annotator.cc]: Ah, I see you also did that change, thank you very much :)

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**